apps/project/serializers.py

```diff
@@ -959,3 +959,21 @@ def create(self, validated_data):
 
     def update(self, instance, validated_data):
         return super().update(instance, validated_data)
+
+
+class UserProjectLeaveSerializer(ProjectPropertySerializerMixin, serializers.Serializer):
+
+    def validate(self, data):
+        membership = ProjectMembership.objects.filter(
+            project=self.project,
+        )
+        if membership.count() == 1:
+            raise serializers.ValidationError('Cannot leave the project')
+        if membership.filter(role=ProjectRole.get_owner_role()).count() == 1:
+            raise serializers.ValidationError('Cannot leave the project')
+        return data
```

Comment:
We need to update this as done in this PR:https://github.com/the-deep/server/pull/1541#pullrequestreview-2480895241 Or rebase this with that branch.